### DISCUSSION TITLE
How to get the Bluetooth working ?

### DISCUSSION THREAD

**Discussion by pbranly** [Blackymas/NSPanel_HA_Blueprint#2087] [Q&A]
Hi I have added the line in the configuration, recompiled and now I have a Bluetooth icon along the WiFi one. 
But I don't see any additional sensor coming from esphome. 
Do I have to make additional settings ? 
Thanks 
Phil

**Comment by edwardtfn** [comment]
If you've added Bluetooth Proxy add-on, the system will automatically proxy the sounrounding compatible BLE devices to Home Assistant, so it will only be visible as Bluetooth in Home Assistant, like if you have a remote Bluetooth antena. Please read more here: https://esphome.io/components/bluetooth_proxy.html

If you are talking about BLE tracker, then you have to add some setting as indicated here: https://esphome.io/components/esp32_ble_tracker

**Reply by pbranly** [reply]
Hi
I have a Bluetooth line in my HA configuration 
So I see the Bluetooth of the NUC in which HA is running
But I don't see the Bluetooth interface coming from the nspanel
I surely miss something !
Phil

**Reply by edwardtfn** [reply]
The Bluetooth proxy is almost transparent, so you won't see as entities, devices or integrations in Home Assistant, but it will be forwarding the messages from a nearby BLE device to Home Assistant. If you have a BLE device communicating to HA, that device will be able to use either your Bluetooth integration or the Bluetooth proxy, so BLE devices far from your HA antenna but close to your panel will be able to use it to communicate to HA.

**Reply by Doughboy68** [reply]
Do we really have to recompile the first time using serial? I had everything all edited and was ready to compile OTA and then noticed the little note on webpage. If so, a suggesting is to make that requirement more noticeable at the beginning of the article since I just about missed this.

`The first time this component is enabled for an ESP32, the code partition needs to be resized. Please flash the ESP32 via USB when adding this to your configuration. After that, you can use OTA updates again.`

https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/addon_bluetooth_proxy.md

And it is also a bit ambiguous to me if we have to add/uncomment these lines... other examples online have these but is this already set within the nspanel custom configuration?

```
# esp32_ble_tracker:
#   scan_parameters:
#     interval: 1100ms
#     window: 1100ms
#     active: true
#
# bluetooth_proxy:
#   active: true

```

**Reply by Doughboy68** [reply]
Update... I did one of my panels, already using esp-idf so I just tried OTA and is working.
With latest tft etc. there is a bluetooth icon now above the wifi on the panel.
I can also confirm my iphone setup in Private BLE tracker is being picked up by the panel based on its source MAC address.

All I did was uncomment out the line for bluetooth proxy, nothing else in my yaml config.
Upgrading my other panels now.

**Reply by edwardtfn** [reply]
That is a recommendation from ESPHome docs, but I've also updated with OTA only without any issue.

**Reply by pbranly** [reply]
Hi how do you set your iPhone in « private ble tracker «  ?
Thanks
Phil

**Reply by Doughboy68** [reply]
You need to get IRK for device.  
https://www.home-assistant.io/integrations/private_ble_device

**Reply by pbranly** [reply]
Hi
So I need a Mac for that ? I don’t have ! 
phil

**Reply by Doughboy68** [reply]
https://community.home-assistant.io/t/guide-getting-the-irk-for-iphone-and-apple-watch-without-macos-maybe-also-other-devices/623694

**Reply by pbranly** [reply]
Thank you . I have a spare esp32 and will try but this seems to be very tricky and reserved for a wide minority ! 
phil

**Reply by Doughboy68** [reply]
You can also setup the ESPresense which may be easier. This is how I did mine.
Once program ESP32 device with ESPresence, go to http://192.168.10.185/ui/#/  (that IP will be different for you)
Click Enroll to pair the Apple device using the web gui in the ESP32 device and then it will show IRK in MQTT using mqtt explorer.
https://espresense.com/beacons/apple

You can remove after once you get the IRK unless you want to continue to track with ESPresence then you can place the IRKs in the Known BLE resolving keys.

https://esphome.io/components/binary_sensor/ble_presence.html
With the esp32_ble_tracker, you have to include all your BLE devices that wish to track in the ESP32 config for each device and compile, which seems unrealistic to do for certain use cases, where other methods like ESPresence or Private BLE it is configured/tracked with Home assistant. Maybe it is possible to have NSPanel do something all on its own if a particular BLE device comes in range, or have HA do something based on sensor. But if the BLE MAC or IRK changes you have to recompile again.

The ESPresence I linked to above uses mqtt to communicate with Home Assistant. The Private BLE is an intergration in HA that is just for Apple Devices and you require the IRK to use. I personally don't really track things with BLE, I just setup to learn/figure out for curiosity but I did not find it reliable to track room to room and really have no need. I use BLE for things like inexpensive temperature/humidity monitors which work quite well.  I have several of these around the house  https://ca.govee.com/products/govee-bluetooth-hygrometer-thermometer-h5075
Oh and my toothbrush connects also but I have never found a use for that really.

**Reply by plasterer41** [reply]
Is it possible to use the Nspanel Bluetooth to act like espresense - how would I load the MQTT settings and topic onto the Nspanel?

Any advice would be greatly received :-)

**Comment by pbranly** [comment]
> Update... I did one of my panels, already using esp-idf so I just tried OTA and is working. With latest tft etc. there is a bluetooth icon now above the wifi on the panel. I can also confirm my iphone setup in Private BLE tracker is being picked up by the panel based on its source MAC address.
> 
> All I did was uncomment out the line for bluetooth proxy, nothing else in my yaml config. Upgrading my other panels now.



> I can also confirm my iphone setup in Private BLE tracker is being picked up by the panel based on its source MAC address.

Hi how do you set your iPhone in «  private ble tracker » ? 
thanks
Phil

**Reply by plasterer41** [reply]
are you trying to add your phone (is it an iphone) to Espresense?

**Comment by AbeltjeNL** [comment]
@edwardtfn, is there a way to get the Bluetooth Proxy add-on working without flashing the device via serial/usb cable?
